### PR TITLE
mount tunnel: use PidRef

### PR DESCRIPTION
**Pull request by bluca**

### CODE REVIEW

File: src/shared/mount-util.c [hunk 1]
```diff
@@ -1087,24 +1087,29 @@ static int mount_in_namespace(
...
 
-        assert(target > 0);
         assert(propagate_path);
         assert(incoming_path);
         assert(src);
         assert(dest);
         assert(!options || is_image);
 
-        r = namespace_open(target, &pidns_fd, &mntns_fd, NULL, NULL, &root_fd);
+        if (!pidref_is_set(target))
```
Comment (YHNdnzj) [inline src/shared/mount-util.c:1096]: Hmm, maybe keep this as an assertion?

File: src/shared/mount-util.c [hunk 2]
```diff
@@ -1087,24 +1087,29 @@ static int mount_in_namespace(
...
 
-        r = in_same_namespace(target, 0, NAMESPACE_MOUNT);
+        r = in_same_namespace(target->pid, 0, NAMESPACE_MOUNT);
         if (r < 0)
                 return log_debug_errno(r, "Failed to determine if mount namespaces are equal: %m");
         /* We can't add new mounts at runtime if the process wasn't started in a namespace */
         if (r > 0)
                 return log_debug_errno(SYNTHETIC_ERRNO(EINVAL), "Failed to activate bind mount in target, not running in a mount namespace");
 
+        if (pidref_verify(target) < 0)
```
Comment (YHNdnzj) [inline src/shared/mount-util.c:1110]: Most of the times it doesn't matter, but this can fail for other reasons.

**Review by YHNdnzj — APPROVED** [review]
minor stuff really

File: src/shared/mount-util.c (hunk 1 above)
Comment (bluca) [reply src/shared/mount-util.c:1096]: Other uses of PidRef never assert, just run this same check

File: src/shared/mount-util.c [hunk 3]
```diff
@@ -1087,24 +1087,29 @@ static int mount_in_namespace(
...
+                return log_debug_errno(SYNTHETIC_ERRNO(ESRCH), "Failed to verify target process '" PID_FMT "': %m", target->pid);
```
Comment (YHNdnzj) [inline src/shared/mount-util.c:1111]: This still always logs synthetic ESRCH

**Pull request merged**